src/python/WMCore/WMSpec/WMWorkloadTools.py

```diff
@@ -215,7 +215,7 @@ def validateSiteLists(arguments):
         whiteList = [whiteList]
     if not isinstance(blackList, list):
         blackList = [blackList]
-    res = (set(whiteList) & set(blackList))
+    res = [s for s in list(set(whiteList) & set(blackList)) if s]
```

Comment:
On  0, Eric Vaandering <notifications@github.com> wrote:
> > @@ -215,7 +215,7 @@ def validateSiteLists(arguments):
> >          whiteList = [whiteList]
> >      if not isinstance(blackList, list):
> >          blackList = [blackList]
> > -    res = (set(whiteList) & set(blackList))
> > +    res = [s for s in list(set(whiteList) & set(blackList)) if s]
> 
> Ah, thanks. I understand now. Lines 215 and 217 are the real problem. Could add "and blackList" and else blackList = []

yes, exactly. And, it is matter of preferences, either we need to modify
215/217 lines as you suggested or define res as I propose. In addition,
Seangchan should fix code which pass arguments here and properly json'ise
arguments.